Sources/performance-worker/BenchmarkClient.swift

```diff
@@ -80,18 +89,123 @@ struct BenchmarkClient {
     }
   }
 
+  private func computeTimeAndErrorCode<Contents>(
```

Comment:
It's a bit strange to link the timing with the capturing of error codes. Aim for simple: have a function do one thing; time the execution of the body and return the result:

```swift
func timeIt<R>(_ body: () async throws -> R) async rethrows -> (R, seconds: Double) {
  // ...
}
```

This means at the call site you can return through the error code if applicable:

```swift
let (errorCode, seconds) = await timeIt { 
  do {
    try await client.whatever() 
    return nil
  } catch let error as RPCError {
    return error.code
  } catch {
    return .unknown
  }
}
```